sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,230 @@
+    """"""
+
+    cert_path: Optional[StrictStr]
+    """"""
+
+    key_path: Optional[StrictStr]
+    """"""
+
+    ssl_password: Optional[StrictStr]
+    """"""
+
+
+class HazelcastOnlineStore(OnlineStore):
+    _client: Optional[HazelcastClient] = None
+
+    def _get_client(self, config: HazelcastOnlineStoreConfig):
+        logger = logging.getLogger("hazelcast")
+        logger.setLevel(logging.ERROR)
+        if not self._client:
+            if config.cloud:
+                self._client = HazelcastClient(
+                    cluster_name=config.cluster_name,
+                    statistics_enabled=True,
+                    ssl_enabled=True,
+                    cloud_discovery_token=config.discovery_token,
+                    ssl_cafile=config.ca_path,
+                    ssl_certfile=config.cert_path,
+                    ssl_keyfile=config.key_path,
+                    ssl_password=config.ssl_password
+                )
+            elif config.ssl_enabled:
+                self._client = HazelcastClient(
+                    cluster_name=config.cluster_name,
+                    statistics_enabled=True,
+                    ssl_enabled=True,
+                    ssl_cafile=config.ca_path,
+                    ssl_certfile=config.cert_path,
+                    ssl_keyfile=config.key_path,
+                    ssl_password=config.ssl_password
+                )
+            else:
+                self._client = HazelcastClient(
+                    statistics_enabled=True,
+                    cluster_members=config.cluster_members,
+                    cluster_name=config.cluster_name
+                )
+        return self._client
+
+    @log_exceptions_and_usage(online_store="hazelcast")
+    def online_write_batch(
+            self,
+            config: RepoConfig,
+            table: FeatureView,
+            data: List[
+                Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+            ],
+            progress: Optional[Callable[[int], Any]]
+    ) -> None:
+
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+        project = config.project
+
+        query = f"""
+            SINK INTO {_feature_map_name(project, table)}
+            (entity_key, feature_name, feature_value, event_ts, created_ts)
+            values (?, ?, ?, ?, ?);
+        """
+        for entity_key, features, timestamp, created_ts in data:
+            entity_key_bin = serialize_entity_key(
+                entity_key,
+                entity_key_serialization_version=2,
+            ).hex()
+            timestamp = _convert_to_naive_utc(timestamp)
+            if created_ts is not None:
+                created_ts = _convert_to_naive_utc(created_ts)
+            for feature_name, feature_value in features.items():
+                client.sql.execute(query,
```

Comment:
Put operation is done via SQL query since we are using JSON format as value. I don't know if simple `map.put()` operation is faster. Any comments?